apps/settings/js/modules/date_time.js

```diff
@@ -320,7 +314,7 @@ define(function(require) {
         return d.toLocaleFormat(format);
       } else {
         return d.toLocaleString(navigator.languages, {
-          hour12: this._currentHour12,
+          hour12: navigator.mozHour12,
```

Comment:
cannot. navigator.mozHour12 may be false in which case navigator.hour12 will be picked.

On the bright side, we don't do that anywhere else, so if we'll ever standardize it (I doubt it would be in this form) we'll have to update it everywhere.